Tutorials/WinForms/Classic/implement-crud-functionality-with-xpo-objects.md

```diff
@@ -0,0 +1,169 @@
+* Adjust buttons' positions and sizes.
+* Rename the `simpleButton1` control to **btnReload** and set the `Text` property to **&Reload**.
+* Double click the `btnReload` control to add the `Click` event handler (you can use the **Properties** window to add it).
+* Select all lines in the `EditCustomerForm_Load` method and click the **Edit > Refactor > Extract Method** menu item or use **Ctrl+R,Ctrl+M**.
+* Change the method name to **Reload** and call this method in the `btnReload_Click` event handler.
+* Modify the `btnSave_Click` method.
+    ```csharp
+    using DevExpress.Xpo.DB.Exceptions;
+    // ...
+    private void btnSave_Click(object sender, EventArgs e) {
+        try {
+            UnitOfWork.CommitChanges();
+            Close();
+        } catch (LockingException) {
+            XtraMessageBox.Show(this, "The record was modified or deleted by another user. Please click the Reload button and try again.", "XPO Tutorial", MessageBoxButtons.OK, MessageBoxIcon.Stop);
+        }
+    }
+    ```
+## Create a new record
+  * Open the `CustomersListForm` in the designer.
+  * Click the [smart-tag](https://docs.microsoft.com/en-us/dotnet/framework/winforms/controls/performing-common-tasks-using-smart-tags-on-wf-controls) glyph.
+  * Select the **Convert to RibbonForm** menu item.
+  * Change the `ribbonPage1` text to **Home**.
+  * Change the `ribbonPageGroup1` text to **Edit**.
+  * Delete the `ribbonPageGroup2`.
+  * Add the `BarButtonItem` item to the **Edit** group and change its name to **btnNew**.
+  * Set the `btnNew.Caption` property to **New**.
+  * Double-click the **New** button to create an event handler.
+  * Select the `using` statement in the `CustomersGridView_RowClick` method and click the **Edit > Refactor > Extract Method** menu item (**Ctrl+R,Ctrl+M**).
+  * Change the method name to **ShowEditForm**.
+  * Change the `customerID` parameter type to `int?`.
+  * Call the `ShowEditForm` method in the `btnNew_ItemClick` event handler and pass the `null` value as a parameter.
+  * A new object initially does not have an identifier. XPO assigns a value to it from the auto-incremented key column when a new object is saved to the database. Open the `EditCustomerForm` code and add the following line to the `btnSave_Click` method after the `CommitChanges` method call:
+    ```csharp
+    CustomerID = ((Customer)CustomerBindingSource.DataSource).Oid;
+    ```
+  * Run the application, click the New button, fill editors, and click the **Save** button to see the result.
+## Delete a selected record
+  * Open the `CustomersListForm` and add the **Delete (btnDelete)** button to the `RibbonControl`.
+  * Double-click the **Delete** button to add the event handler.
+  * Change the `Reload` method to make the `Session` instance available in the class scope:
```

Comment:
"in the class scope"?

Reply:
I changed this part to better explain what to do and why.
63ae5f54ed52c25de10cc71f24d33113fb181f3e